### PR TITLE
Canvas api.js

### PR DESCRIPTION
**Pull request by YashKumarVerma**
Partial of #905 

Code Refactor to follow ES6 standards

Fixes 533 Issues

### CODE REVIEW

File: public/js/canvasApi.js [hunk 1]
```diff
@@ -1,259 +1,258 @@
...
-    else if (dir == "UP")
-        return [y1, -x1];
-    else
-        return [x1, y1];
+    ctx.rect(
+        Math.round(globalScope.ox + xx + x1 - correction) + correction,
+        Math.round(globalScope.oy + yy + y1 - correction) + correction,
+        Math.round(x2),
+        Math.round(y2),
+    );
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/canvasApi.js]: hound not accepting trailing comma, 
p.s. hound was supposed to not check .js files.

File: public/js/canvasApi.js [hunk 2]
```diff
@@ -1,259 +1,258 @@
...
+    x1 *= globalScope.scale;
+    y1 *= globalScope.scale;
+    x2 *= globalScope.scale;
+    y2 *= globalScope.scale;
+    ctx.rect(
+        Math.round(globalScope.ox + x1 - correction) + correction,
+        Math.round(globalScope.oy + y1 - correction) + correction,
+        Math.round(x2),
+        Math.round(y2),
+    );
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/canvasApi.js]: hound not accepting trailing comma, 
p.s. hound was supposed to not check .js files.

File: public/js/canvasApi.js [hunk 3]
```diff
@@ -1,259 +1,258 @@
...
-      pi = Math.PI;
-    ctx.arc(Math.round(xx + globalScope.ox + Sx +correction)-correction, Math.round(yy + globalScope.oy + Sy + correction)-correction, Math.round(radius), newStart + pi, newStop + pi);
+    if (counterClock) pi = Math.PI;
+    ctx.arc(
+        Math.round(xx + globalScope.ox + Sx + correction) - correction,
+        Math.round(yy + globalScope.oy + Sy + correction) - correction,
+        Math.round(radius),
+        newStart + pi,
+        newStop + pi,
+    );
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/canvasApi.js]: hound not accepting trailing comma, 
p.s. hound was supposed to not check .js files.

File: public/js/canvasApi.js [hunk 4]
```diff
@@ -1,259 +1,258 @@
...
     [newStart, newStop, counterClock] = rotateAngle(start, stop, dir);
-    ctx.arc(Math.round(xx + globalScope.ox + Sx+correction)-correction, Math.round(yy + globalScope.oy + Sy+correction)-correction, Math.round(radius), newStart, newStop, counterClock);
+    ctx.arc(
+        Math.round(xx + globalScope.ox + Sx + correction) - correction,
+        Math.round(yy + globalScope.oy + Sy + correction) - correction,
+        Math.round(radius),
+        newStart,
+        newStop,
+        counterClock,
+    );
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/canvasApi.js]: hound not accepting trailing comma, 
p.s. hound was supposed to not check .js files.

File: public/js/canvasApi.js [hunk 5]
```diff
@@ -1,259 +1,258 @@
...
+    xx *= globalScope.scale;
+    yy *= globalScope.scale;
+    ctx.bezierCurveTo(
+        Math.round(xx + ox + x1),
+        Math.round(yy + oy + y1),
+        Math.round(xx + ox + x2),
+        Math.round(yy + oy + y2),
+        Math.round(xx + ox + x3),
+        Math.round(yy + oy + y3),
+    );
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/canvasApi.js]: hound not accepting trailing comma, 
p.s. hound was supposed to not check .js files.

File: public/js/wire.js [hunk 6]
```diff
@@ -1,166 +1,172 @@
...
+    // to update the state of wire
+    update() {
+        let updated = false;
+        if (embed) return updated;
+
+        if (this.node1.absX() === this.node2.absX()) {
+            this.x1 = this.node1.absX();
+            this.x2 = this.node1.absX();
+            this.type = 'vertical';
+        } else if (this.node1.absY() === this.node2.absY()) {
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/wire.js]: difference between x1 and x2

File: public/js/wire.js [hunk 7]
```diff
@@ -1,166 +1,172 @@
...
-        return true;
-    else if ((this.type == "horizontal") && (this.node1.absX() > this.node2.absX()) && (x < this.node1.absX()) && (x > this.node2.absX()) && (y === this.node2.absY()))
-        return true;
-    else if ((this.type == 'vertical') && (this.node1.absY() < this.node2.absY()) && (y > this.node1.absY()) && (y < this.node2.absY()) && (x === this.node2.absX()))
-        return true;
-    else if ((this.type == 'vertical') && (this.node1.absY() > this.node2.absY()) && (y < this.node1.absY()) && (y > this.node2.absY()) && (x === this.node2.absX()))
-        return true
-    return false;
+    checkWithin(x, y) {
+        if (this.type === 'horizontal' && y === this.node2.absY()) {
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/wire.js]: difference between x1 and x2 / node1 and node2

File: public/js/canvasApi.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tachyons) [reply public/js/canvasApi.js]: Rebase with master

**Pull request closed without merging**